presto-tests/src/test/java/io/prestosql/execution/EventsBuilder.java

```diff
@@ -16,20 +16,36 @@
 import com.google.common.collect.ImmutableList;
 import io.prestosql.spi.eventlistener.QueryCompletedEvent;
 import io.prestosql.spi.eventlistener.QueryCreatedEvent;
+import io.prestosql.spi.eventlistener.QueryMetadata;
 import io.prestosql.spi.eventlistener.SplitCompletedEvent;
 
 import java.util.List;
 import java.util.concurrent.CountDownLatch;
 import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+
+import static com.google.common.base.Predicates.alwaysTrue;
+import static java.util.Objects.requireNonNull;
 
 class EventsBuilder
 {
+    private final Predicate<QueryMetadata> queryFilter;
     private ImmutableList.Builder<QueryCreatedEvent> queryCreatedEvents;
     private ImmutableList.Builder<QueryCompletedEvent> queryCompletedEvents;
     private ImmutableList.Builder<SplitCompletedEvent> splitCompletedEvents;
 
     private CountDownLatch eventsLatch;
 
+    public EventsBuilder()
+    {
+        this(alwaysTrue());
+    }
+
+    public EventsBuilder(Predicate<QueryMetadata> queryFilter)
+    {
+        this.queryFilter = requireNonNull(queryFilter, "filter is null");
```

Comment:
part of the setup (construction) is done in `initialize` and part is done now in ctor.
let's merge the two.

let's rename `initialize` to `reset` and let's call `reset(0)` in the ctor, so that newly created object is usable